metric_learn/sdml.py

```diff
@@ -112,7 +112,11 @@ def _fit(self, pairs, y):
                                                   return_inverse=True)
     diff = pairs[:, 0] - pairs[:, 1]
     loss_matrix = (diff.T * y).dot(diff)
-    emp_cov = prior_inv + self.balance_param * loss_matrix
+    emp_cov = (prior_inv + self.balance_param * loss_matrix +
+               # We add a small value on the diagonal in case the
+               # emp_cov matrix is singular (see
+               # #https://github.com/metric-learn/metric-learn/issues/202)
+               np.eye(diff.shape[1]) * 1e-10)
```

Comment:
This value makes the test pass, I think it's a value small enough to not interfere with the other coeffs, what do you think ?

Reply:
I'll revert it given the strategy we'll adopt (see https://github.com/metric-learn/metric-learn/pull/203#pullrequestreview-237083152)